packages/x-components/src/x-modules/tagging/service/pdp-add-to-cart.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Store } from 'vuex';
+import { Result } from '@empathyco/x-types';
+import { BrowserStorageService, StorageService } from '@empathyco/x-storage-service';
+import { Logger, logger } from '@empathyco/x-logger';
+import { RootXStoreState } from '../../../store/index';
+import { XPlugin } from '../../../plugins/index';
+import { PDPAddToCartService } from './types';
+
+/**
+ * Default implementation for the {@link PDPAddToCartService}.
+ *
+ * @public
+ */
+export class DefaultPDPAddToCartService implements PDPAddToCartService {
+  protected logger: Logger;
+  /**
+   * Session id key to use as key in the storage.
+   *
+   * @public
+   */
+  public static readonly RESULT_CLICKED_ID_KEY = 'add-to-cart';
+
+  /**
+   * Global instance of the {@link PDPAddToCartService}.
+   */
+  public static instance: PDPAddToCartService = new DefaultPDPAddToCartService();
+
+  public constructor(
+    protected localStorageService: StorageService = new BrowserStorageService(localStorage, 'x'),
+    protected sessionStorageService: StorageService = new BrowserStorageService(sessionStorage, 'x')
+  ) {
+    this.logger = logger.child(`[PDPAddToCartService]`);
+  }
+
+  protected get store(): Store<RootXStoreState> {
+    return XPlugin.store;
+  }
+
+  storeResultClicked(result: Result): void {
+    const { clickedResultStorageKey, clickedResultStorageTTLMs } =
+      this.store.state.x.tagging.config;
+
+    let key = result[clickedResultStorageKey as keyof Result] as string;
+
+    if (clickedResultStorageKey === 'url') {
+      key = key.replace(/\s|\+/g, '%20');
+    }
+
+    const storageKey = `${DefaultPDPAddToCartService.RESULT_CLICKED_ID_KEY}-${key}`;
+    this.localStorageService.setItem(storageKey, result, clickedResultStorageTTLMs as number);
+  }
+
+  moveToSessionStorage(id: string): void {
+    let clickedResultStorageKeyId: string;
+
+    if (id === 'url') {
+      const url = window.location.href.replace(/\s|\+/g, '%20');
+      clickedResultStorageKeyId = `${DefaultPDPAddToCartService.RESULT_CLICKED_ID_KEY}-${url}`;
+    } else {
+      clickedResultStorageKeyId = `${DefaultPDPAddToCartService.RESULT_CLICKED_ID_KEY}-${id}`;
+    }
+
+    const result = this.localStorageService.getItem(clickedResultStorageKeyId);
+    if (result) {
+      this.sessionStorageService.setItem(clickedResultStorageKeyId, result);
+      this.localStorageService.removeItem(clickedResultStorageKeyId);
+    }
+  }
+
+  trackAddToCart(id?: string | null): void {
+    let clickedResultStorageKeyId: string;
+
+    if (!id) {
```

Comment:
Considering the productId is coming directly from the snippet or the API (which is fed by the customer), maybe we should make a safer validation? 
Like for example, if we receive an empty string, it will be `!'' === true` but should be treated as if the id is not provided.

Reply:
I'm not sure if I follow you, if the customer provides and empty string there, that is a falsy value and it will be treated as no id was provided, which I think is the correct way, isn't it? 😅